GSA-TTS/FAC
# Error Pages

# Error Pages

Issue: https://github.com/GSA-TTS/FAC/issues/1750
Figma: [Link](https://www.figma.com/file/lnkXJTfKzD3YBhiJZ2QFEv/FAC-Source-of-Truth?node-id=1%3A23921&mode=dev)

## Changes:

Error pages! New pages 400, 401, 404, 500, and updated page 403. 

## How to test:

1. Switch to this branch
2. Ensure `DEBUG` always evaluates to `False` in your `config/settings.py` (line ~212). Otherwise, error pages don't display and you'll get a stacktrace. 
    * I do this by setting the `DJANGO_DEBUG` environment variable to False. 
3. Run normally, and navigate to some pages that throw errors.
    * For 404, navigate to [`localhost:8000/anything-you-want-al-long-as-it-doesnt-exist`](http://localhost:8000/anything-you-want-al-long-as-it-doesnt-exist)
    * For 403, navigate to an audit that isn't yours, such as: [`http://localhost:8000/audit/submission-progress/ABUNCHOFNUMBERSNSTUFF`](http://localhost:8000/audit/submission-progress/ABUNCHOFNUMBERSNSTUFF)
    * The others (400, 401, 500) are tougher to get on purpose. Just to test them out, you can change a view temporarily to `return render(request, "401.html")` before doing anything else. Navigating to the relevant page will then _show_ the error page. 

## Screenshots:
400:
![image](https://github.com/GSA-TTS/FAC/assets/91098850/880aff4f-f4f9-426d-9f71-8efef829bd69)
401:
![image](https://github.com/GSA-TTS/FAC/assets/91098850/b170a1bc-95a3-4e23-8f41-af82d8322168)
403:
![image](https://github.com/GSA-TTS/FAC/assets/91098850/99445517-4eed-4ec5-998b-6e91af484a1a)
404: 
![image](https://github.com/GSA-TTS/FAC/assets/91098850/ee7381bb-a440-4c75-8ff0-fd33b1ce6cd7)
500:
![image](https://github.com/GSA-TTS/FAC/assets/91098850/954b1c7b-6efa-47a0-a439-0f500ca44f0a)

## PR checklist: submitters


## PR checklist: reviewers

        
The larger the PR, the stricter we should be about these points.
